sympy/core/evalf.py

```diff
@@ -149,6 +149,20 @@ def fastlog(x: Optional[MPF_TUP]) -> tUnion[int, Any]:
     return x[2] + x[3]
 
 
+def fastlog10_for_expr(v: 'Expr'):
+    """
+    Quickly compute a positive integer m such that 10^m > |v|, for any
+    :py:class:`~.Expr` *v*.
+    """
+    # Do `.evalf(1)` before taking `abs`, or else `abs` can be very slow.
+    # Taking `.evalf(1)` can make the value smaller in magnitude, but we make
+    # up for it by adding 1 at the end.
+    a = abs(v.evalf(1))
+    b = int(math.ceil(a)).bit_length()
+    c = math.ceil(b / LG10)
```

Comment:
It looks like `math.ceil(abs(v))` will call `evalf` implicitly:
```python
In [9]: x = 10**100*pi

In [10]: math.ceil(x).bit_length()
Out[10]: 334
```
Why do we need `fastlog10` instead of `fastlog` above which uses base 2?

Reply:
The `.evalf(1)` makes it faster, and is good enough for the approximation we want here.

```python
f = cyclotomic_poly(5)
v = CRootOf(f, 0)
timeit.timeit('v._reset(); math.ceil(abs(v))', number=10, globals=globals())
Out[7]: 10.512659683999999
timeit.timeit('v._reset(); math.ceil(abs(v.evalf(1)))', number=10, globals=globals())
Out[8]: 0.9664351640000035
```

Reply:
On `fastlog10` versus `fastlog`: I discuss below, where it comes up again.